pkg/app/csi-snapshot.go

```diff
@@ -0,0 +1,226 @@
+	return true, nil
+}
+
+func (csi *CSISnapshot) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "deployment",
+		Name:      csi.name,
+		Namespace: csi.namespace,
+	}
+}
+
+func (csi *CSISnapshot) Uninstall(ctx context.Context) error {
+	err := csi.cli.CoreV1().Namespaces().Delete(ctx, csi.namespace, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("Namespace deleted successfully", field.M{"app": csi.name})
+	return nil
+}
+
+func (csi *CSISnapshot) Ping(ctx context.Context) error {
+	log.Print("Pinging the application.", field.M{"app": csi.name})
+
+	listDirectories := []string{"sh", "-c", "ls /var/log"}
+	stderr, err := csi.execCommand(ctx, listDirectories)
+	if err != nil {
+		return errors.Wrapf(err, "Error while Pinging the application %s", stderr)
+	}
+
+	log.Print("Ping to the application was success.", field.M{"app": csi.name})
+	return nil
+}
+
+func (csi *CSISnapshot) GetClusterScopedResources(ctx context.Context) []crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (csi *CSISnapshot) Count(ctx context.Context) (int, error) {
+	return 3, nil
+}
+
+func (csi *CSISnapshot) ConfigMaps() map[string]crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (csi *CSISnapshot) Secrets() map[string]crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (csi *CSISnapshot) execCommand(ctx context.Context, command []string) (string, error) {
+	podname, containername, err := kube.GetPodContainerFromDeployment(ctx, csi.cli, csi.namespace, csi.name)
+	if err != nil || podname == "" {
+		return "", errors.Wrapf(err, "Error getting pod and containername %s.", csi.name)
+	}
+	_, stderr, err := kube.Exec(csi.cli, csi.namespace, podname, containername, command, nil)
+	return stderr, err
+}
+
+func (csi CSISnapshot) getAppDeploymentObj() (*appsv1.Deployment, error) {
+	deploymentManifest := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: time-logger
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: time-logger
+  template:
+    metadata:
+      labels:
+        app: time-logger
+    spec:
+      containers:
+      - name: test-container
+        image: ghcr.io/kanisterio/kanister-tools:0.72.0
+        command: ["sh", "-c"]
+        args: ["while true; do for x in $(seq 1200); do date >> /var/log/time.log; sleep 1; done; truncate /var/log/time.log --size 0; done"]
+        volumeMounts:
+        - name: data
+          mountPath: /var/log
+      volumes:
+      - name: data
+        persistentVolumeClaim:
+          claimName: time-log-pvc`
+
+	var deployment *appsv1.Deployment
+	err := yaml.Unmarshal([]byte(deploymentManifest), &deployment)
+	return deployment, err
+}
+
+func (csi CSISnapshot) getAppPersistentVolumeClaimObj() (*v1.PersistentVolumeClaim, error) {
+	pvcManifest := `apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: time-log-pvc
+  labels:
+    app: time-logger
+spec:
+  storageClassName: csi-hostpath-sc
```

Comment:
@shlokchaudhari9 I somehow think that we should change the `csi-hostpath-sc` to default storage class and then use that automatically instead of manually specifying the storage class here.
What I have concluded is, the other database applications are somehow not working with `csi-hostpath-sc`, can we figure that out? If we can figure out why those are not working with `csi-hostpath-sc`, we can set this as default storage class and then we will not have to set the storeageclass anywhere else.

Reply:
@viveksinghggits This is the issue I am facing on all the database apps https://github.com/kubernetes/minikube/issues/13098

The problem is that, database apps in our tests are being run by non-root users (some helm charts explicitly forbid running their containers as root) That's why we get permission denied error in most of them or helm chart gets timed-out during installation.

I also found out why only CSISnapshot (Time logger) integration test passes, while other apps fail. Turns out, the image used in time-logger deployment is being run as a root user. Hence, we do not get permission denied error when using the mount path within the container.

In the attached issue thread, many people have suggested switching the volume provisioner. Which is what we are doing by not setting the hostpath storageClass as default. We run the CSISnapshot (Time logger) integration test explicitly by mentioning the storageClassName in PVC, while the other database apps automatically use standard volume provisioner.

Reply:
makes sense. We can get this in.

Reply:
I will re-review.

Reply:
Cool Thanks